### PR TITLE
test predict for new and cold users

### PR DESCRIPTION
**Pull request by monkey0head**
tests added to check:
- models, which are able to recommend for cold/new users, really do it
- models, which are not able to recommend for cold/new users filter those users out and do not fail

bugs found and fixed:
- cluster model did not filter users in `_predict`
- in neuromf net.forward could return a number instead of tensor for batch of one element as `.squeeze` was used without `dim` parameter

### CODE REVIEW

File: tests/models/test_all_models.py [hunk 1]
```diff
@@ -223,3 +232,115 @@ def test_filter_seen(log):
         log=log, users=["user1"], k=5, filter_seen_items=False
     )
     assert pred.count() == 4
+
+
+def get_kwargs(model, user_features):
```
Comment (AleXXL1986) [inline tests/models/test_all_models.py:237]: what does the method make?
not so good name

**Review by AleXXL1986 — CHANGES_REQUESTED** [review]

File: tests/models/test_all_models.py (hunk 1 above)
Comment (monkey0head) [reply tests/models/test_all_models.py:237]: it is a function which returns dict with kwargs to pass during models fit and predict.
for models, which use features, it adds user features to kwargs. it helps to run test for all models, whether features are  used or not.

Comment (monkey0head) [reply tests/models/test_all_models.py:237]: deleted

File: tests/models/test_all_models.py [hunk 2]
```diff
@@ -223,3 +232,115 @@ def test_filter_seen(log):
...
+def test_predict_cold_and_new_filter_out(model, long_log_with_features):
+    pred = fit_predict_selected(
+        model,
+        train_log=long_log_with_features.filter(sf.col("user_id") != "u1"),
+        inf_log=long_log_with_features,
+        user_features=None,
+        users=["u1", "cold_user"],
+    )
+    # assert new/cold users are filtered out in `predict`
+    if isinstance(model, LightFMWrap) or not model.can_predict_cold_users:
```
[the remark this answers is not in the document]
Comment (monkey0head) [reply tests/models/test_all_models.py:343]: if we pass user features, lightFM can predict cold users. 
In this test we do not pass user features and look if the model just does not fail and just do not return result for cold/new users.

**Review by AleXXL1986 — APPROVED** [review]

**Pull request merged**